go/enclave/rollupchain/rollup_chain.go

```diff
@@ -679,16 +679,17 @@ func (rc *RollupChain) GetBalance(encryptedParams common.EncryptedParamsGetBalan
 func (rc *RollupChain) signRollup(r *obscurocore.Rollup) {
 	var err error
 	h := r.Hash()
-	r.Header.R, r.Header.S, err = ecdsa.Sign(rand.Reader, rc.privateKey, h[:])
+	r.Header.R, r.Header.S, err = ecdsa.Sign(rand.Reader, rc.enclavePrivateKey, h[:])
 	if err != nil {
 		log.Panic("Could not sign rollup. Cause: %s", err)
 	}
 }
 
 func (rc *RollupChain) verifySig(r *obscurocore.Rollup) bool {
+	// Todo - it is not enough to verify the signature. We have to also check that the signer is one of the attested aggregators.
 	h := r.Hash()
 	if r.Header.R == nil || r.Header.S == nil {
 		panic("Missing signature on rollup")
 	}
-	return ecdsa.Verify(&rc.privateKey.PublicKey, h[:], r.Header.R, r.Header.S)
+	return ecdsa.Verify(&rc.enclavePrivateKey.PublicKey, h[:], r.Header.R, r.Header.S)
```

Comment:
I'm still confused about how this doesn't break haha. If this is verifying a rollup from a different enclave then its own public key won't work will it?

Or are all the enclave's private keys the same at the moment? Or is it always the same enclave publishing the rollups?

Reply:
it doesn't. lol. I wasn't checking the result of the verification